src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1021 @@ const ActorInit En_Js_InitVars = {
     (ActorFunc)EnJs_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_8096ABA0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_ENEMY, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_ENEMY,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 20, 40, 0, { 0, 0, 0 } },
 };
 
-#endif
+static Gfx* D_8096ABCC[] = {
+    gMoonChildMajorasMaskDL, gMoonChildOdalwasMaskDL,   gMoonChildGohtsMaskDL,
+    gMoonChildGyorgsMaskDL,  gMoonChildTwinmoldsMaskDL,
+};
+
+static f32 D_8096ABE0[] = { 1.0f, 0.5f, 0.5f, 0.48f, 0.45f };
+
+static f32 D_8096ABF4[] = { 60.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+
+static f32 D_8096AC08[] = { -450.0f, 1400.0f, 1470.0f, 1670.0f, 1470.0f };
+
+static f32 D_8096AC1C[] = { 0.0f, 700.0f, 900.0f, 900.0f, 900.0f };
+
+static Vec3f D_8096AC30 = { 500.0f, -500.0f, 0.0f };
+
+void EnJs_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    s16 cs;
+    s32 i;
+    EnJs* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.01f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gMoonChildSkel, &gMoonChildStandingAnim, this->jointTable,
+                       this->morphTable, MOONCHILD_LIMB_MAX);
+    Animation_PlayLoop(&this->skelAnime, &gMoonChildStandingAnim);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->actionFunc = func_8096971C;
+    this->maskType = 0;
+    this->actor.terminalVelocity = -9.0f;
+    this->actor.gravity = -1.0f;
+
+    cs = this->actor.cutscene;
+    i = 0;
+    while (i != 2) {
+        this->cutscenes[i] = cs;
+        if (cs != -1) {
+            this->actor.cutscene = cs;
+            cs = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        }
+        i++;
+    }
+
+    this->cutsceneIndex = -1;
+
+    switch (ENJS_GET_TYPE(&this->actor)) {
+        case 0:
+            this->maskType = 0;
+            this->actionFunc = func_8096A6F4;
+
+            Animation_PlayLoop(&this->skelAnime, &gMoonChildSittingAnim);
+            func_8016566C(0x3C);
+
+            if (gSaveContext.save.weekEventReg[84] & 0x20) {
+                Inventory_DeleteItem(ITEM_MASK_FIERCE_DEITY, SLOT(ITEM_MASK_FIERCE_DEITY));
+                gSaveContext.save.weekEventReg[84] &= (u8)~0x20;
+                break;
+            }
+        default:
+            break;
```

Comment:
This seems odd. Could you just put a break at the end of case 0, and have the default at the bottom, instead of a fallthrough?